pytorch_toolkit/object_detection/face-detection/tools/eval.py

```diff
@@ -0,0 +1,257 @@
+        for line in content:
+            if line.startswith(beginning):
+                average_precisions.append(float(line.replace(beginning, '')))
+    return average_precisions
+
+
+def get_sha256(path, work_dir):
+    """ Gets sha256. """
+
+    os.makedirs(work_dir, exist_ok=True)
+    os.system(f'sha256sum {path} > {work_dir}/sha256.txt')
+    with open(f'{work_dir}/sha256.txt') as read_file:
+        sha256 = read_file.readlines()[0].strip().split(' ')[0]
+    return sha256
+
+
+def get_size(path, work_dir):
+    """ Gets size of a file. """
+
+    os.makedirs(work_dir, exist_ok=True)
+    os.system(f'ls -l {path} > {work_dir}/ls.txt')
+    with open(f'{work_dir}/ls.txt') as read_file:
+        size = read_file.readlines()[0].strip().split(' ')[4]
+    return int(size)
+
+
+def compute_wider_metrics(config_path, snapshot, work_dir, wider_dir, outputs):
+    """ Computes WiderFace metrics on easy, medium, hard subsets. """
+
+    wider_data_folder = wider_dir
+    os.makedirs(wider_data_folder, exist_ok=True)
+
+    wider_data_zip = os.path.join(wider_data_folder, 'WIDER_val.zip')
+    assert os.path.exists(wider_data_zip), f'failed to find WIDER_val.zip here: {wider_data_zip}'
+    subprocess.run(f'unzip -q -o {wider_data_zip} -d {wider_data_folder}'.split(' '), check=True)
+
+    eval_tools_zip = os.path.join(wider_data_folder, 'eval_tools.zip')
+    if not os.path.exists(eval_tools_zip):
+        subprocess.run(
+            f'wget http://shuoyang1213.me/WIDERFACE/support/eval_script/eval_tools.zip'
+            f' -O {eval_tools_zip}'.split(' '), check=True)
+    subprocess.run(f'unzip -q -o {eval_tools_zip} -d {wider_data_folder}'.split(' '), check=True)
+
+    wider_annotation_zip = os.path.join(wider_data_folder, 'ider_face_split.zip')
+    if not os.path.exists(wider_annotation_zip):
+        subprocess.run(
+            f'wget http://mmlab.ie.cuhk.edu.hk/projects/WIDERFace/support/bbx_annotation/wider_face_split.zip'
+            f' -O {wider_annotation_zip}'.split(' '), check=True)
+    subprocess.run(f'unzip -q -o {wider_annotation_zip} -d {wider_data_folder}'.split(' '), check=True)
+
+    wider_annotation = os.path.join(wider_dir, 'wider_face_split', 'wider_face_val_bbx_gt.txt')
+    wider_images = os.path.join(wider_dir, 'WIDER_val', 'images')
+    wider_coco_annotation = os.path.join(wider_dir, 'instances_val.json')
+    subprocess.run(
+        f'python {FACE_DETECTION_TOOLS}/wider_to_coco.py'
+        f' {wider_annotation} {wider_images} {wider_coco_annotation}'.split(' '), check=True)
+
+    res_pkl = os.path.join(work_dir, 'wider_face_res.pkl')
+    config_with_wider_face = os.path.join(work_dir, 'config_with_wider_face.py')
+    os.system(f'cp {config_path} {config_with_wider_face}')
+    replace_text_in_file(config_with_wider_face, 'ann_file=', f'ann_file="{wider_coco_annotation}",#')
+    replace_text_in_file(config_with_wider_face, 'img_prefix=', f'img_prefix="{wider_dir}",#')
```

Comment:
Maybe there's a way to override some of the config file parameters in a more natural way, rather than replacing text in config code? I'm just a bit concerned that current approach might be fragile.

Reply:
fixed